Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1458,7 +1458,15 @@ private extension ChatChannelController {
     }
 
     private func getFirstUnreadMessageId() -> MessageId? {
-        guard let lastReadMessageId = channel?.reads.first(where: { $0.user.id == client.currentUserId })?.lastReadMessageId else {
+        let messages = self.messages.filter {
+            $0.type == .regular || $0.type == .reply || $0.type == .deleted
+        }
+
+        let currentUserUnread = channel?.reads.first(where: {
+            $0.unreadMessagesCount > 0 && $0.user.id == client.currentUserId
+        })
```

Comment:
When getting the `lastReadMessageId`, we actually need to make sure the message is currenlty unread, otherwise we need to ignore it